include/zephyr/net/net_compat.h

```diff
@@ -0,0 +1,303 @@
+/** @file
+ * @brief Network namespace compatibility mode header
+ *
+ * Allows to use relevant network symbols without the "net_", "NET_" or
+ * "ZSOCK_" prefixes.
+ */
+
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_NET_NET_COMPAT_H_
+#define ZEPHYR_INCLUDE_NET_NET_COMPAT_H_
+
+#if defined(CONFIG_NET_NAMESPACE_COMPAT_MODE)
```

Comment:
I wonder should this ifdef be in include/zephyr/net/net_ip.h instead, and 
should `CONFIG_NET_NAMESPACE_COMPAT_MODE` start being `default y` always?

Then The POSIX_API can include this header on its own from wherever it needs, and CONFIG_NET_NAMESPACE_COMPAT_MODE be switched to `default n` and deprecated(?)